### PR TITLE
Retry on 403 for S3 put in certain environments

### PR DESCRIPTION
**Pull request by ywangd**
This PR configures a new retry condition for s3 client so that it retries on 403 for operations such as PUT in certain environments. Note that 403 is already retried for GET due to S3RetryingInputStream.

Resolves: ES-9321

### CODE REVIEW

**Comment by ywangd** [comment]
Mark the PR as draft since I am not entirely convinced this is useful enough. The extra retries may help when the credentials finishing progagation before the retries are exhuasted. This may not be very realistic. I have observed GET failures in cluster logs that happened around the same time when PUT failed. GET already retries on 403 due to S3RetryingInputStream and they still failed after 4 attempts.

File: modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3ClientSettingsTests.java [hunk 1]
```diff
@@ -207,12 +207,18 @@ public void testMaxConnectionsCanBeSet() {
...
         assertThat(defaultConfiguration.getMaxConnections(), is(ClientConfiguration.DEFAULT_MAX_CONNECTIONS));
-        ClientConfiguration configuration = S3Service.buildConfiguration(settings.get("other"));
+        ClientConfiguration configuration = S3Service.buildConfiguration(settings.get("other"), false);
         assertThat(configuration.getMaxConnections(), is(maxConnections));
 
         // the default appears in the docs so let's make sure it doesn't change:
         assertEquals(50, ClientConfiguration.DEFAULT_MAX_CONNECTIONS);
     }
+
+    public void testNonDefaultRetryPolicy() {
```
Comment (henningandersen) [inline modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3ClientSettingsTests.java:219]: Should we call thist `testStatelessDefaultRetryPolicy`?

**Review by henningandersen — APPROVED** [review]
LGTM.

File: modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3ClientSettingsTests.java (hunk 1 above)
Comment (ywangd) [reply modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3ClientSettingsTests.java]: Sure pushed 5103ae4b655ed42dd69d895223053fb03efec6dc

**Comment by ywangd** [comment]
@elasticmachine update branch

**Pull request merged**